### PR TITLE
Re-enabling and fixing the BytesReader tests

### PR DESCRIPTION
**Pull request by shiftylogic**
From #1419

### CODE REVIEW

**Comment by shiftylogic** [comment]
@ahsonkhan

File: tests/System.Buffers.Experimental.Tests/BytesReader.cs [hunk 1]
```diff
@@ -10,13 +10,14 @@ namespace System.Slices.Tests
...
     {
-        [Fact(Skip = "System.InvalidProgramException : Common Language Runtime detected an invalid program.")]
+        [Fact]
         public void SingleSegmentBytesReader()
         {
             ReadOnlyBytes bytes = Create("AB CD#EF&&");
             var reader = new BytesReader(bytes);
 
             var ab = reader.ReadBytesUntil((byte)' ');
+            Assert.True(ab.HasValue);
```
Comment (ahsonkhan) [inline tests/System.Buffers.Experimental.Tests/BytesReader.cs]: Why is this necessary?

Comment (shiftylogic) [inline tests/System.Buffers.Experimental.Tests/BytesReader.cs]: It isn't really necessary. I added it during my investigation. However, it is a correct test but mostly benign.
